public/video-ui/src/components/Header.js

```diff
@@ -1,36 +1,86 @@
 import React from 'react';
 import {Link, IndexLink} from 'react-router';
 import VideoSearch from './VideoSearch/VideoSearch';
+import {isVideoPublished} from '../util/isVideoPublished';
+import VideoPublishBar from './VideoPublishBar/VideoPublishBar';
 
 export default class Header extends React.Component {
 
-  render () {
+  publishVideo = () => {
+    this.props.publishVideo(this.props.video.id);
+  };
+
+  renderVideoPublishedInfo() {
```

Comment:
minor: maybe worth making these explicit components. Lets do this in a new PR though.